### ISSUE TITLE
Popover: Component type is missing

### ISSUE THREAD

**Issue by BenJackGill** [primefaces/primevue#6134]
### Describe the bug

I am migrating from V3 to V4.

As per the [migration docs](https://primevue.org/guides/migration/v4/) the `OverlayPanel` has been renewed to `Popover`.

Previously I was using a template ref for the `OverlayPanel` with TypeScript support like this:

```Vue
<template>
    <OverlayPanel ref="overlayPanelElement" />
</template>

<script setup lang="ts">
import type OverlayPanel from "primevue/overlaypanel";
const overlayPanelElement = ref<OverlayPanel | null>(null); // We now have a fully typed OverlayPanel template ref
</script>
```

But when switching to the renamed `Popover` that solution no longer works:

```Vue
<template>
    <Popover ref="popoverElement" />
</template>

<script setup lang="ts">
import type Popover from "primevue/popover";
const popoverElement = ref<Popover | null>(null); // <---- TYPESCRIPT ERROR SHOWS HERE
</script>
```

It now throws this TypeScript error:

> 'Popover' refers to a value, but is being used as a type here. Did you mean 'typeof Popover'?ts(2749)

One workaround is to use this, but it's a hassle and bad ergonomics:

```Vue
const popoverElement = ref<InstanceType<typeof Popover> | null>(null);
```

Would love to have the types again please.

Note: I have provided a Stackblitz but the TypeScript error is not visible there. See screenshot below for the IDE error:

![Screenshot 2024-07-29 at 10 14 19 AM](https://github.com/user-attachments/assets/d1cc0ce6-6568-45e5-8790-aa18bbcf0ebc)

### Reproducer

https://stackblitz.com/edit/primevue-4-ts-vite-issue-template-z5eubn?file=src%2FApp.vue

### PrimeVue version

4.0.2

### Vue version

4.x

### Language

TypeScript

### Build / Runtime

Nuxt

**Comment by BenJackGill** [comment]
I'm guessing the problem is more widespread, but I have also found the same issue with these other components:

- `DatePicker` (previously named `Calendar` in v3)
- `AutoComplete`

**Comment by doug-shontz** [comment]
Agreed.  Thank you for the workaround, but, as a new user of PrimeVue, this drove me crazy for a bit trying to figure out.
